lockfile/src/python.rs

```diff
@@ -33,7 +33,10 @@ fn is_requirements_file(path: &Path) -> bool {
 impl Parse for PyRequirements {
     /// Parses `requirements.txt` files into a vec of packages
     fn parse(&self, data: &str) -> anyhow::Result<Vec<Package>> {
-        let (_, entries) = pypi::parse(data)
+        // Ignore newlines with line continuation characters.
+        let input = data.replace(" \\\n", " ").replace(" \\\r\n", " ");
```

Comment:
This is absolutely atrocious and I hate that I'm even suggesting this. But the only alternative to this is to rewrite the entire main loop of the Python parser to not split on ` \\\n` (probably by trying to use nom for this), then pass these full strings to our parsing functions and replace `space0` and `space1` with a function that treats ` \\\n` as normal whitespace.

At least that's the only alternative I could come up with and it wasn't something I'd like to implement to fix this issue (though maybe I should?).